src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3132,7 +3132,7 @@ final void endMessage() throws SQLServerException {
     // the client MUST send the next packet with both ignore bit (0x02) and EOM bit (0x01)
     // set in the status to cancel the request.
     final boolean ignoreMessage() throws SQLServerException {
-        if (packetNum > 0) {
+        if (packetNum > 0 || this.tdsMessageType == TDS.PKT_BULK) {
```

Comment:
we usually do ( 0 < packetNum || TDS.PKT_BULK == this.tdsMessageType)

Reply:
This only make sense for `==`, not to confuse it with assignment `=`, so I only updated my change.